feat: create user profile mockup page

PanisaBD opened the pull request:
##   What’s Added
- Mockup screen for user profile page
- Adding new features (without backend logic) includes 
   - profile image
   - fiends image + name
   - edit profile button
   - logout button
   - friend search form/button
   - add friend button

##   Notes
- This is a UI-only mockup (no real data or backend logic yet)

##   Related Commits
- feat: create user profile mockup page (sorry for the lack of commit na)

MildlyBrain commented on src/app/profile/page.tsx:
```diff
@@ -0,0 +1,336 @@
+            }}
+          >
+            <Image
+              src="https://placehold.co/108x108"
```
Using an outside source doesn't work out-of-the-box since it's not configured to gather sources from a remote host (which leads to [this error](https://nextjs.org/docs/messages/next-image-unconfigured-host)).

MildlyBrain commented on src/app/profile/page.tsx:
```diff
@@ -0,0 +1,336 @@
+import BackgroundWithNoise from "@/components/BackgroundWithNoise";
+import Image from "next/image";
+
+const styleVars = {
+  "--color-dark-blue": "#41415F",
+  "--color-dark-gray": "#545456",
+  "--color-red": "#A7271D",
+  "--color-coral": "#C57564",
+  "--color-yellow": "#F4D590",
+  "--color-white": "#FFFFFF",
+} as React.CSSProperties;
```
Have you got the `globals.css` working yet? 
I'm not sure I can settle with an in-page constant of colors.
This also includes all the custom variables used throughout the page.

MildlyBrain commented on src/app/profile/page.tsx:
```diff
@@ -0,0 +1,336 @@
+"use client";
+
+import React from "react";
```
I don't think you need to import `React`.

PanisaBD commented:
##  What’s Changed
- Refactored inline styles to Tailwind CSS
- Use global.css variables instead of in-page constant
- Separated UI into reusable components includes :
   - Back Button
   - Edit Button
   - Log Out Button
   - Page Action
   - Profile Section
   - Friend List
   - Search Box
- Replaced placeholder image with avatar from /public/images

## Known Issues
- font-bai is not applied correctly 
- Bottom page action area shape is not rendering properly

MildlyBrain commented on src/app/profile/page.tsx:
```diff
@@ -0,0 +1,63 @@
+      <div className="font-bai bg-orange min-h-screen text-base">
+        <div
+          className="min-h-screen p-6 flex flex-col items-center gap-6 relative overflow-hidden"
+          style={{
+            background:
+              "linear-gradient(180deg, var(--color-red) 7.32%, var(--color-orange) 47.03%)",
+          }}
```
This could be done using TailwindCSS too! It has [a class specifically for this](https://tailwindcss.com/docs/background-image#adding-a-linear-gradient).
Also, the gradient starts from 0-100% (I asked @SiravitPhokeed last night).

MildlyBrain commented on src/components/BackButton.tsx:
```diff
@@ -0,0 +1,27 @@
+  return (
+    <button
+      onClick={() => router.back()}
+      className={`absolute top-10 left-4 w-11 h-11 flex items-center justify-center ${bgColorClass}`}
```
The `cn` helper could be useful here!

MildlyBrain commented on src/components/BackButton.tsx:
```diff
@@ -0,0 +1,27 @@
+      className={`absolute top-10 left-4 w-11 h-11 flex items-center justify-center ${bgColorClass}`}
+      type="button"
+    >
+      <span className={`font-icon ${iconColorClass} text-icon-size`}>
```
Also here.

MildlyBrain commented on src/components/BottomBanner.tsx:
What is this used for?
Also, use TailwindCSS instead of inline styles.

MildlyBrain commented on src/components/PageAction.tsx:
Have you got a solution for the Page Action not being rounded properly?
If not, I'll try to help you here (keep in mind I might not be able to do it too).

PanisaBD replied:
Thanks, I’ve already figured out the solution — might include the fix in my next push.

MildlyBrain reviewed (changes requested):
Additionally from the in-line comments.

- Use text sizes from the global variables. These are also defined in the UI specs.
- Might be a bit much to ask, but I'd suggest using a plain `Button` component, then extending that component later for non-trivial buttons. 
  - You can wait for when we merge @NlNJ4 's PR, then start extending (and editing) your Buttons from there. Of course, you may modify his base Button component too if you think it's not flexible enough.
  - For buttons that appear on almost every page (like BackButton), that's okay to make it another component to make it less of a hassle to use repeatedly, although the BackButton itself should use `Button` component too.
  - If every button is a component, then the point of having components (which is reusability) is not really there, is it? 
- Per our `colocation` in the coding style guide, do consider moving profile-page-only components to a `components` folder right in the `/app/profile`

SiravitPhokeed commented:
While you wait for my review (sorry about that by the way), I quickly made a **comparison shot** so you can make some tweaks in the meantime.

![](https://github.com/user-attachments/assets/d1fa1daf-66fb-47da-a446-6e87e8fdccb1)

Some things that immediately jumped out at me:
- The **whole page sits a little too high**, so not enough to padding maybe.
- The **back Button is squished** (not square) because the icon didn’t have enough height (might need more margin/padding).
- The **full name text** should get more horizontal padding so it wraps nicely. Throw in a `text-balance`, too.
- I think a better **approach to the Friend Grid** would be to keep the gap consistent but scale the profile picture to always fill the column.

There might be more I miss so stay vigilant haha. But a lot of things look pretty good already, if the code has no issues I’ll let this merge.

SiravitPhokeed reviewed (approved):
It’s…good enough. Far from perfect, but mergeable.

You are encouraged to look into the issues I mentioned in the previous comment but it’s not the end of the world if we merge this.
